tarfs.go

```diff
@@ -100,39 +101,54 @@ func (fs TarWriter) Close() error {
 // a catar).
 type TarReader struct {
 	r *gnutar.Reader
+	root *File
 }
 
-var _ FilesystemReader = TarReader{}
+var _ FilesystemReader = &TarReader{}
 
 // NewTarFS initializes a new instance of a GNU tar archive that can be used
 // for catar archive tar/untar operations.
-func NewTarReader(r io.Reader) TarReader {
-	return TarReader{gnutar.NewReader(r)}
+func NewTarReader(r io.Reader, addRoot bool) *TarReader {
```

Comment:
Could you add a new struct `TarReaderOptions` and put the bool into that one? That would make it more flexible if more additional fields have to be added later. Take a look at https://github.com/folbricht/desync/blob/master/localfs.go#L53 and https://github.com/folbricht/desync/blob/master/cmd/desync/untar.go#L18